src/backend/distributed/commands/vacuum.c

```diff
@@ -77,23 +78,14 @@ PostprocessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
 	CitusVacuumParams vacuumParams = VacuumStmtParams(vacuumStmt);
 	const char *stmtName = (vacuumParams.options & VACOPT_VACUUM) ? "VACUUM" : "ANALYZE";
 
-	/*
-	 * No table in the vacuum statement means vacuuming all relations
-	 * which is not supported by citus.
-	 */
-	if (list_length(vacuumStmt->rels) == 0)
-	{
-		/* WARN for unqualified VACUUM commands */
-		ereport(WARNING, (errmsg("not propagating %s command to worker nodes", stmtName),
-						  errhint("Provide a specific table in order to %s "
-								  "distributed tables.", stmtName)));
-	}
-
+	bool vacuum_all = false;
 
 	List *citusRelationIdList = VacuumCitusRelationIdList(vacuumStmt, vacuumParams);
-	if (list_length(citusRelationIdList) == 0)
```

Comment:
I think we should keep this.
This is for when `VACUUM` refers to some tables that are not distributed.

Reply:
Actually, we are checking if we have any distributed table as below instead of list length check.

`citusRelationIdList = VacuumCitusRelationIdList(vacuumStmt, vacuumParams);

bool distributedVacuumStmt = IsDistributedVacuumStmt(vacuumParams.options,
														citusRelationIdList);
if (!distributedVacuumStmt)
{
	return NIL;
}`